### PR TITLE
Contact journal extension duration picker (EXPOSUREAPP-5110)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Duration picker for contact journal.

How to test:
1) Go to test menu
2) Contact Diary Test Options
3) Click on "00:00" text field on top of the screen
4) Select any duration different from "00:00"
5) Check that text filed changed its value, changed color to black and now has blue border
6) Open dialog again
7) Select zero duration (00:00)
8) Check that text is now gray and don't have blue border

Vuzix View_20210218_215906_670

### CODE REVIEW

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/contactdiary/ui/ContactDiaryTestFragment.kt [hunk 1]
```diff
@@ -42,6 +47,29 @@ class ContactDiaryTestFragment : Fragment(R.layout.fragment_test_contact_diary),
...
             locationVisitsCleanButton.setOnClickListener { vm.clearLocationVisits() }
             personEncountersCleanButton.setOnClickListener { vm.clearPersonEncounters() }
+            durationValue.setOnClickListener {
+                val args = Bundle()
+                args.putString(ContactDiaryDurationPickerFragment.DURATION_ARGUMENT_KEY,
+                    binding.durationValue.text.toString())
+
+                val durationPicker = ContactDiaryDurationPickerFragment()
+                durationPicker.arguments = args
+                durationPicker.setTargetFragment(this@ContactDiaryTestFragment, 0)
```
Comment (mtwalli) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/contactdiary/ui/ContactDiaryTestFragment.kt:57]: We should not use this, could you please use `navbackstackentry`
https://developer.android.com/guide/navigation/navigation-programmatic#navbackstackentry

File: Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <NumberPicker
+            android:id="@+id/hours"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="67dp"
+            app:layout_constraintBottom_toBottomOf="@+id/divider"
+            app:layout_constraintEnd_toStartOf="@+id/divider"
+            app:layout_constraintHorizontal_bias="0.5"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="@+id/divider" />
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml:32]: Divider is grey for NumberPicker, I think it should be blue afair right?

File: Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        <NumberPicker
+            android:id="@+id/minutes"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="18dp"
+            android:layout_marginEnd="67dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.5"
+            app:layout_constraintStart_toEndOf="@+id/divider"
+            app:layout_constraintTop_toBottomOf="@+id/title" />
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml:59]: same here?

File: Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        <android.widget.Button
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml:61]: why `<android.widget.` ?

File: Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="29dp"
+            android:layout_marginBottom="@dimen/spacing_small"
+            android:text="@string/duration_dialog_cancel_button"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/ok_button"
+            app:layout_constraintTop_toBottomOf="@+id/minutes" />
+
+        <android.widget.Button
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml:74]: why `<android.widget.` ?

File: Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml [hunk 6]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/contact_diary_duration_picker_dialog_fragment.xml:4]: Please remove this <layout><layout>

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        with(binding.minutes) {
+            minValue = 0
+            maxValue = minutesArray.size - 1
+            displayedValues = minutesArray
+        }
+
+        with(binding) {
+            val duration = requireArguments().getString(DURATION_ARGUMENT_KEY)!!.split(":").toTypedArray()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt:43]: please use navArgs() I think it works with dialogs right?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+import org.joda.time.format.PeriodFormatterBuilder
+
+class ContactDiaryDurationPickerFragment : DialogFragment() {
+
+    interface OnChangeListener {
+        fun onChange(duration: Duration)
+    }
+
+    private var _binding: ContactDiaryDurationPickerDialogFragmentBinding? = null
+    private val binding get() = _binding!!
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt:20]: don't we have a lazy for that ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        with(binding) {
+            val duration = requireArguments().getString(DURATION_ARGUMENT_KEY)!!.split(":").toTypedArray()
+            hours.value = hoursArray.indexOf(duration[0])
+            minutes.value = minutesArray.indexOf(duration[1])
+
+            cancelButton.setOnClickListener { dismiss() }
+            okButton.setOnClickListener {
+                (targetFragment as? OnChangeListener)?.onChange(getDuration(hours.value, minutes.value))
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt:49]: we would not need this with `navbackstackentry`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    companion object {
+        const val DURATION_ARGUMENT_KEY = "duration"
+
+        val minutesArray = arrayOf("00", "15", "30", "45")
+        val hoursArray = Array(24) {
+            if (it < 10) {
+                "0$it"
+            } else {
+                it.toString()
+            }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/durationpicker/ContactDiaryDurationPickerFragment.kt:69]: val hoursArray = Array(24) { "%02d".format(it) }?

**Review by mtwalli — APPROVED** [review]

**Comment by axelherbstreith** [comment]
Looking good 👍

**Review by axelherbstreith — APPROVED** [review]

**Pull request merged**